rustls/src/crypto/mod.rs

```diff
@@ -371,6 +371,28 @@ pub trait SupportedKxGroup: Send + Sync + Debug {
     /// This can fail if the random source fails during ephemeral key generation.
     fn start(&self) -> Result<Box<dyn ActiveKeyExchange>, Error>;
 
+    /// Start and complete a key exchange, in one operation.
+    ///
+    /// The default implementation for this calls `start()` and then calls
+    /// `complete()` on the result.  This is suitable for Diffie-Hellman-like
+    /// key exchange algorithms, where there is not a data dependency between
+    /// our key share (named "pub_key" in this API) and the peer's (`peer_pub_key`).
+    ///
+    /// If there is such a data dependency (like key encapsulation mechanisms), this
+    /// function should be implemented.
+    fn start_and_complete(&self, peer_pub_key: &[u8]) -> Result<CompletedKeyExchange, Error> {
```

Comment:
Revisiting [semver guidance](https://doc.rust-lang.org/cargo/reference/semver.html#trait-new-default-item) this is "Possibly-breaking", but only if the same method exists in another trait that would be implemented on a type implementing `CryptoProvider`, and that feels unlikely :+1: